cmd/orphan-finder/main.go

```diff
@@ -119,15 +130,23 @@ func parseLogLine(sa certificateStorage, logger blog.Logger, line string) (found
 	// backdated we need to add the backdate duration to find the true issued
 	// time.
 	issuedDate := cert.NotBefore.Add(backdateDuration)
-	_, err = sa.AddCertificate(ctx, der, int64(regID), nil, &issuedDate)
+	ocspResponse, err := ca.GenerateOCSP(ctx, core.OCSPSigningRequest{
+		CertDER: der,
+		Status:  string(core.OCSPStatusGood),
+	})
+	if err != nil {
+		logger.AuditErrf("Couldn't generate OCSP: %s, [%s]", err, line)
+		return true, false
+	}
+	_, err = sa.AddCertificate(ctx, der, int64(regID), ocspResponse, &issuedDate)
```

Comment:
I think doing this here is mostly fine, but one thing to think about is that this mostly replicates the private CA method `generateOCSPAndStoreCertificate`. It might be nice, either now or in a follow-up change, to instead make that method public and call that RPC here. That way we wouldn't be in danger of this code becoming stale and out of sync with how we generate OCSP/store certificates in the traditional issuance path.

Reply:
(This would also allow us to reduce the gRPC clients the tool requires to just the CA one.)

Reply:
I'm not super keen on exporting another method from the CA. I like keeping it clean and simple. I think the amount of code duplication is fairly minimal here and unlikely to fall out of sync.